src/Instance.ts

```diff
@@ -291,8 +291,12 @@ export class Marked {
 
       if (opt.hooks) {
         opt.hooks.options = opt;
+        opt.hooks.block = blockType;
       }
 
+      const lexer = opt.hooks ? opt.hooks.provideLexer() : (blockType ? _Lexer.lex : _Lexer.lexInline);
+      const parser = opt.hooks ? opt.hooks.provideParser() : (blockType ? _Parser.parse : _Parser.parseInline);
```

Comment:
I may be misunderstanding here: This falls back to default lexer and parser if no hook is provided. But the default `provideLexer` and `provideParser` hooks also perform a fallback to the default lexer and parser. Is one of these fallbacks redundant?

Reply:
Yes technically the default hooks are redundant, that is so someone can just provide one hook and the rest of the hooks provide the default behavior. Currently hooks is null by default so we don't have to call the functions unnecessarily if no hooks are changed.